base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -257,11 +257,21 @@ where T: BlockchainBackend
             validators,
             config,
         };
-        if blockchain_db.get_height()?.is_none() {
+        let metadata = blockchain_db.get_metadata()?;
+        if metadata.height_of_longest_chain.is_none() {
             let genesis_block = consensus_manager.get_genesis_block();
             blockchain_db.store_new_block(genesis_block)?;
             blockchain_db.store_pruning_horizon(config.pruning_horizon)?;
+        } else if ((metadata.pruning_horizon == 0) && (metadata.pruning_horizon != config.pruning_horizon)) |
+            ((config.pruning_horizon != 0) && (config.pruning_horizon < metadata.pruning_horizon))
```

Comment:
Should be `||`

Reply:
Think this can be simplified
```suggestion
        } else if config.pruning_horizon > 0 && (metadata.pruning_horizon == 0 || config.pruning_horizon < metadata.pruning_horizon)
```

"If the config says pruned mode and either the node was in archive mode or the config horizon is less than the current horizon, then update the pruning horizon"

Reply:
There was a small logic bug that I fixed. The code should also be much more readable now. The logic should have been: If an archival node then apply configured pruning horizon, or if a pruned node apply configured pruning horizon if less.  The last case might change at a later stage.